UI/CUIControls.h

```diff
@@ -594,4 +595,48 @@ class RotatingGraphic : public GG::StaticGraphic {
     float   m_phase_offset;
 };
 
+class CUIToggleButton;
+
+/** \brief A FreeOrion styled ToggleButton. */
+class CUIToggleButtonRepresenter : public GG::StateButtonRepresenter {
+public:
+    virtual void   Render(const CUIToggleButton& button) const;
+
+    virtual void   OnChecked(bool checked) const;
+
+    virtual GG::Pt MinUsableSize(const CUIToggleButton& button) const;
+};
+
+/** \brief Toggled StateButton
+ * 
+ * This StateButton renders a SubTexture depending on the checked state.
+ * With the default single texture it renders the same icon recolored towards grey.
+ * When a second SebTexture is assigned, it will render the appropriate SubTexture.
+ */
+class CUIToggleButton : public GG::StateButton {
```

Comment:
briefly, references to textures.

From what I can understand, representers are designed for rendering a specific state, but not to hold any specific references.  Initially I had all of the render logic here, but while the buttons worked and displayed tooltip, I was not  able to produce a call to CUIToggleButtonRepresenter->Render().

Should I look into restructuring TechTreeWnd to utilize a representer?  This appeared to add more complexity since the representer needs a reference to a specific button anyway, but I am obviously missing something crucial here.